tests/integration/source_code/solacc.py

```diff
@@ -90,22 +90,38 @@ def lint_all(file_to_lint: str | None):
             continue
         _missing_imports, _parseable = lint_one(file, ctx, unparsed)
         for _import in _missing_imports:
-            count = missing_imports.get(_import, 0)
-            missing_imports[_import] = count + 1
+            register_missing_import(missing_imports, _import)
         parseable += _parseable
     all_files_len = len(all_files) - (len(skipped) if skipped else 0)
     parseable_pct = int(parseable / all_files_len * 100)
+    missing_imports_count = sum(sum(details.values()) for details in missing_imports.values())
     logger.info(
-        f"Skipped: {len(skipped or [])}, parseable: {parseable_pct}% ({parseable}/{all_files_len}), missing imports: {sum(missing_imports.values())}"
+        f"Skipped: {len(skipped or [])}, parseable: {parseable_pct}% ({parseable}/{all_files_len}), missing imports: {missing_imports_count}"
     )
-    missing_imports = dict(sorted(missing_imports.items(), key=lambda item: item[1], reverse=True))
-    for key, value in missing_imports.items():
-        logger.info(f"Missing import '{key}': {value} occurrences")
+    log_missing_imports(missing_imports)
     # fail the job if files are unparseable
     if parseable_pct < 100:
         sys.exit(1)
 
 
+def register_missing_import(missing_imports: dict[str, dict[str, int]], missing_import: str):
+    prefix = missing_import.split(".")[0]
+    details = missing_imports.get(prefix, None)
+    if details is None:
+        details = {}
+        missing_imports[prefix] = details
+    count = details.get(missing_import, 0)
+    details[missing_import] = count + 1
```

Comment:
If you want to go pythonic all the way:

```suggestion
def register_missing_import(missing_imports: dict[str, dict[str, int]], missing_import: str):
    prefix = missing_import.split(".")[0]
    details = missing_imports.setdefault(prefix, collections.defaultdict(int))
    details[missing_import] += 1
```